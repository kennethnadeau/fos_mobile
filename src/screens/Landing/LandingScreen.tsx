import React, { useEffect } from "react";
import { View, StyleSheet } from "react-native";
import { ScreenFC } from "react-native-navigation-register-screens";
import Logo from "components/Logo/Logo";
import FOSButton from "components/Button/FOSButton";
import { SCREENS } from "@fos/constants";
import { SafeAreaView } from "react-native-safe-area-context";
import { Text, Button } from "react-native-elements";
import { vs, s, ms } from "react-native-size-matters";
import { useTranslation } from "react-i18next";
import LandingHeaderBg from "@fos/assets/svg/landingHeaderBg.svg";
import { goToOTPScreen } from "@fos/helpers/navigation";
import { Colors } from "@fos/themes";
import RNBootSplash from "react-native-bootsplash";

const LandingScreen: ScreenFC = ({ componentId }) => {
  const { t } = useTranslation(["screens"]);

  useEffect(() => RNBootSplash.hide({ duration: 250 }), []);

  const createNewAccountText = t("Create New Account");
  const loginText = t("Log In");

  return (
    <SafeAreaView style={styles.container}>
      <LandingHeaderBg width="100%" />
      <Text style={styles.headerText}>{t("Data. Meets. People.")}</Text>
      <View style={styles.logoContainer}>
        <Logo />
      </View>
      <View style={styles.buttonContainer}>
        <FOSButton
          accessibilityLabel={createNewAccountText}
          accessible
          containerStyle={styles.createNewAccountBtnContainer}
          onPress={() => goToOTPScreen(componentId)}
          size="large"
          title={createNewAccountText}
          titleStyle={styles.createNewAccountBtnTitle}
        />
        <Button
          accessibilityLabel={loginText}
          accessible
          buttonStyle={styles.loginBtn}
          onPress={() => goToOTPScreen(componentId, true)}
          title={loginText}
          type="clear"
        />
      </View>
    </SafeAreaView>
  );
};

LandingScreen.screenName = SCREENS.LANDING;

const styles = StyleSheet.create({
  buttonContainer: {
    alignItems: "center",
    marginBottom: vs(40),
    marginHorizontal: s(30),
    paddingHorizontal: s(30),
  },
  container: {
    flex: 1,
  },
  createNewAccountBtnContainer: {
    width: ms(300),
  },
  createNewAccountBtnTitle: {
    fontWeight: "700",
  },
  headerText: {
    ...StyleSheet.absoluteFillObject,
    fontSize: s(46),
    fontWeight: "700",
    marginTop: vs(100),
    textAlign: "center",
  },
  loginBtn: { backgroundColor: Colors.transparent },
  logoContainer: {
    alignItems: "center",
    flex: 1,
    justifyContent: "space-around",
    marginTop: vs(5),
  },
});

export default LandingScreen;
