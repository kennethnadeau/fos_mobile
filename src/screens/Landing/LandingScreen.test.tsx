import React from 'react';
import LandingScreen from './LandingScreen';
import {render, RenderAPI} from '@fos/utils/testing';

describe('<LandingScreen />', () => {
  let rendered: RenderAPI;

  beforeEach(() => {
    rendered = render(<LandingScreen componentId="id" />);
  });

  it('should show header', () => {
    const actual = rendered.queryByText('Data. Meets. People.');

    expect(actual).toBeDefined();
  });

  it('should show logo', () => {
    const actual = rendered.queryByA11yLabel('App logo');

    expect(actual).toBeDefined();
  });

  it('should show create new account button', () => {
    const actual = rendered.getByA11yLabel('Create New Account');

    expect(actual).toBeDefined();
  });

  it('should show login button', () => {
    const actual = rendered.getByA11yLabel('Log In');

    expect(actual).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
