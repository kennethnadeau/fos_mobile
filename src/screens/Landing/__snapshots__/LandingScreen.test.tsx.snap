// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingScreen /> should match snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "height": 482.1428571428571,
        "justifyContent": "center",
        "paddingHorizontal": 25.714285714285715,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "testUri": "../../../src/assets/images/landing-banner-bg.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 482.1428571428571,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Object {
          "fontSize": 98.57142857142857,
          "fontWeight": "700",
          "marginTop": 98.08823529411764,
          "textAlign": "center",
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Data. Meets. People.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
        "marginTop": 9.808823529411764,
      }
    }
  >
    <Image
      accessibilityLabel="App logo"
      accessible={true}
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/logo.png",
        }
      }
      style={
        Array [
          Object {
            "height": 270.7235294117647,
            "width": 323.57142857142856,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 78.47058823529412,
        "marginHorizontal": 64.28571428571428,
        "paddingHorizontal": 64.28571428571428,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
          "width": 471.42857142857144,
        }
      }
    >
      <View
        accessibilityLabel="Create New Account"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 64.28571428571428,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 88.27941176470588,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 25.714285714285715,
                "fontWeight": "700",
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Create New Account
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityLabel="Log In"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 19.61764705882353,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#2089dc",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Log In
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
